### ISSUE TITLE
5s stop in merged points

### ISSUE THREAD

**Issue by siferati** [gnoling/UnlinkMKV#24]
I'm trying this tool with Coalgirls release of Nekomonogatari, but the generated mkv always has a big stop in the merged points.

Take the first episode for example:

Episode - 5s stop - Opening - 5s stop - Episode - 5s stop - Ending

I also tried to use the --fixaudio and --fixvideo options, but the result was the same.

Is there a way to solve this?

**Comment by gnoling** [comment]
Can you try the updated dist/ build? If it still gives you issues, please add -ll trace and include the extended output? Coalgirls releases tend to be some of the most problematic.

**Comment by siferati** [comment]
Turns out this wasn't a problem with unlinkmkv, but a limitation of mkvmerge.

Mkvmerge can only split [right before a key frame](https://github.com/mbunkus/mkvtoolnix/wiki/Splitting-imprecise).

The problem was that in the episode I was testing, that keyframe was at 0:01:06.483, but the subtitles started at 0:01:06.48. This meant that mkvmerge couldn't cut the video at this keyframe, since the subtitles had already started, and it can't cut subtitles in half.

So it had to cut at the next available keyframe, which was at 0:01:09.486, resulting in those 3 extra seconds of nothing between the episode and the opening.

I just had to delay the subtitles by one frame to make unlinkmkv work as intendended.

(credits to sneaker from videohelp forum, who helped me find the cause of this problem)
